guizero/Waffle.py

```diff
@@ -60,27 +60,26 @@ def _create_waffle(self):
         self._draw_waffle()
 
     def _create_canvas(self):
-        # if the canvas exists, clear it and destroy it
+        # If the canvas exists, clear it 
         if self._canvas:
             self._canvas.delete("all")
-            self._canvas.destroy()
-
-        # size the canvas. Each element of the waffle has padding to top,
-        # bottom, left and right, but the padding is shared between elements.
-        # What this means in pratice is that elements at the edge have a whole
-        # padding width between them and the edge of the canvas and a single
-        # padding width between individual elements, so the padding between
-        # elements is shared.
-        self._c_height = (self._height * (self._pixel_size + self._pad)) + self._pad
-        self._c_width = (self._width * (self._pixel_size + self._pad)) + self._pad
-
-        # create the canvas and pack it into the waffle frame
-        self._canvas = Canvas(self.tk, height=self._c_height, width=self._c_width, bd=0, highlightthickness=0)
-        self._canvas.pack(fill=BOTH, expand=1)
-
-        # rebind any events as they would have been lost when the canvas
-        # was destroyed
-        self.events.rebind_events(self._canvas)
+        
+        # If not, make one
+        else:
+
+            # size the canvas. Each element of the waffle has padding to top,
+            # bottom, left and right, but the padding is shared between elements.
+            # What this means in pratice is that elements at the edge have a whole
+            # padding width between them and the edge of the canvas and a single
+            # padding width between individual elements, so the padding between
+            # elements is shared.
+            self._c_height = (self._height * (self._pixel_size + self._pad)) + self._pad
+            self._c_width = (self._width * (self._pixel_size + self._pad)) + self._pad
+
+            # create the canvas and pack it into the waffle frame
+            self._canvas = Canvas(self.tk, height=self._c_height, width=self._c_width, bd=0, highlightthickness=0)
```

Comment:
If the waffle is resized at run time (i.e. after the canvas has been created), the canvas size will have to be adjusted to fit the waffle in it. 

I think this is why the `_c_height` and `_c_width` is calculated each time.

Reply:
Sadly I think you're right and this is going to cause more trouble than it fixes. I think this PR should probably be binned for now and I'll think of something else.

Reply:
Is it not possible to change the canvas size at run time? 

e.g. each time the canvas is cleared and setup, it is also sized.

This [article](https://stackoverflow.com/questions/9589867/python-tkinter-change-the-canvas-size-after-inital-declaration)
 suggests you can resize a canvas at run time.

Reply:
Ah, I was more going down the route of how can we nullify the additional enter event. But I'll have a think about that approach instead since I drew a blank yesterday